src/Streamly/Internal/BaseCompat.hs

```diff
@@ -55,3 +57,10 @@ unsafeWithForeignPtr = GHCForeignPtr.unsafeWithForeignPtr
 #else
 unsafeWithForeignPtr = withForeignPtr
 #endif
+
+oneShot :: (a -> b) -> a -> b
+#if MIN_VERSION_base(4,15,0)
```

Comment:
Our build was failing only for 8.0, which means in other version onseShot is present. So we should use min version of base that was shipped with 8.2.